Postgres support

kajuberdut opened the pull request:
Added PostgreSQL support through SQL Alchemy. All tests are currently passing and both the current app and api server run without any obvious errors. This also adds a  config file that will default to debugging on and in-memory SQLite db if no config is found.

kajuberdut committed:
Added empty conftest.py to force pyest to acknowledge the working directory.

kajuberdut committed:
Added db scale test

kajuberdut committed:
Added db scale test

kajuberdut committed:
Linguist (#1)

* Create .gitattributes

Need .gitattributes in order to exclude vendored js files from linguist analysis.

* Update .gitattributes

Fixed paths to vendored files

* Update .gitattributes

* Update .gitattributes

* Update .gitattributes

kajuberdut committed:
Linguist (#1)

* Create .gitattributes

Need .gitattributes in order to exclude vendored js files from linguist analysis.

* Update .gitattributes

Fixed paths to vendored files

* Update .gitattributes

* Update .gitattributes

* Update .gitattributes

kajuberdut committed:
re-write with sqlalchemt

kajuberdut committed:
fixed default db URL

kajuberdut committed:
fixed default db URL

kajuberdut committed:
removed obvious hash example

kajuberdut committed:
Postgres (#2)

* Re-write with sqlalchemy

* Changed default db URL default to SQLite

* Removed overly obvious hash example

kajuberdut committed:
Fixing vendored file list

Specifying which files are vendored by name so that not all static files will be ignored.

kajuberdut committed:
Fixing vendored file list

Specifying which files are vendored by name so that not all static files will be ignored.

kajuberdut committed:
Add sqlalchemy to requirements

kajuberdut committed:
Add sqlalchemy to requirements

kajuberdut committed:
Rebasing to psyprings master

kajuberdut committed:
Rebase from master?

kajuberdut committed:
removing redundant (old) db features.

kajuberdut committed:
Merge branch 'master' of https://github.com/kajuberdut/bottle-wiki into postgres

kajuberdut committed:
Merge branch 'postgres' of https://github.com/kajuberdut/bottle-wiki into postgres

kajuberdut committed:
Fixed module paths and Wikidb init change (db path moved to config).

kajuberdut committed:
Cast datetime to string for json compatability.

kajuberdut committed:
Fixing errors found through tests. Disabling db scale test as it did not achieve what was desired and is timing out.

kajuberdut commented:
A note, we should possibly squash all my comments if we merge.
Also, I had to rebase to get the Travis CI items and am not 100% sure I did that well. Would appreciate comments if artifacts of rebasing are found.

kajuberdut committed:
re-deleting keymail

kajuberdut committed:
removing old non-pytest tests.

mtik00 commented on db/utils.py:
```diff
@@ -0,0 +1,30 @@
+        
+        def getconfig(self, handle):
+                return self.config.get(handle)
+
```
For bonus points, look up `__getitem__`!

kajuberdut replied:
Nifty. Only, now I feel like I should just be subclassing dict to make this config class. Nothing wrong with that I guess.

kajuberdut replied:
Actually... it turns out the standard library has a wonderful configparser. Because of course it does.

kajuberdut committed:
added gitignore entry for vscode launch config.

kajuberdut committed:
switched config file from json to configparser.

kajuberdut committed:
switched config from json to parseconfig. Fixed an incorrect join due to where clause target for v_firstlast. Added simple 'put' to dbapi. Renamed initialize (which initiliazes nothing) to schema.

kajuberdut committed:
see last commit

kajuberdut committed:
Util item @memoized to avoid multiple round trips for same id.
Article tags added to schema.
New Wikidb method "tag" for adding tag to an article.
New Wikidb method "taglist" for listing all tags on an article.
New wikidb method "author" for working with authors (id <-> email) will add entry for new email.
Added taglist to Wikidb method "Detail" output.

kajuberdut committed:
minor fixes and tags api path

kajuberdut committed:
added an api path for create/update article with a json passed by Post.

kajuberdut committed:
added tag handling to json post method in API

kajuberdut committed:
Fixed test_api file and added 2 tests

kajuberdut committed:
tag list order is not intended to be guaranteed so test assertion was wrong. Fixed.

kajuberdut commented:
in Travis CI: py.test is passing, then the lint report is showing up, finally everything is failing with the error:

> make: *** [lint] Error 4

kajuberdut committed:
Fixed items raised by linting (unused import, re-defined from outer scope)
